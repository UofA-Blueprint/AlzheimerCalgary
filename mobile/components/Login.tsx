import { Keyboard, StyleSheet, Text, TextInput, TouchableOpacity, TouchableWithoutFeedback, View } from 'react-native';
import { useState } from 'react';

const Login = () => {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [seePassword, setSeePassword] = useState(true);
    const [checkValidEmail, setCheckValidEmail] = useState(false);
    const [invalidPassword, setInvalidPassword] = useState(false);
    const [invalidPasswordMsg, setInvalidPasswordMsg] = useState('');

    const handleCheckEmail = (text: string) => {
        let re = /\S+@\S+\.\S+/;
        let regex = /^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/im;
    
        setEmail(text);
        if (re.test(text) || regex.test(text)) {
            setCheckValidEmail(false);
        } else {
            setCheckValidEmail(true);
        }
    };

    const handleCheckPassword = (text: string) => {
        setPassword(text)
        const isNonWhiteSpace = /^\S*$/;
        if (!isNonWhiteSpace.test(text)) {
            setInvalidPassword(true)
            setInvalidPasswordMsg('Password must not contain spaces');
            return;
        }
    
        const isContainsUppercase = /^(?=.*[A-Z]).*$/;
        if (!isContainsUppercase.test(text)) {
            setInvalidPassword(true)
            setInvalidPasswordMsg('Password must have at least one uppercase character');
            return;
        }
    
        const isContainsLowercase = /^(?=.*[a-z]).*$/;
        if (!isContainsLowercase.test(text)) {
            setInvalidPassword(true)
            setInvalidPasswordMsg('Password must have at least one lowercase character');
            return;
        }
    
        const isContainsNumber = /^(?=.*[0-9]).*$/;
        if (!isContainsNumber.test(text)) {
            setInvalidPassword(true)
            setInvalidPasswordMsg('Password must contain at least one digit');
            return;
        }
    
        const isValidLength = /^.{8,16}$/;
        if (!isValidLength.test(text)) {
            setInvalidPassword(true)
            setInvalidPasswordMsg('Password must be 8-16 characters long');
            return;
        }

        setInvalidPassword(false)
    };

    const handleLogin = () => {
        if (!invalidPassword) {
            console.log('email:', email)
            console.log('password:', password)
        }
    };

    return (
        <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
            <View style={styles.container}>
                <Text>{}</Text>
                <TextInput
                    style={styles.emailInput}
                    placeholder='Email'
                    value={email}
                    onChangeText={(text) => handleCheckEmail(text)}
                    keyboardType='email-address'
                />

                {checkValidEmail ? (<Text>Invalid email</Text>) : (<Text></Text>)}

                <TextInput
                    style={styles.passwordInput}
                    placeholder='Password'
                    value={password}
                    onChangeText = {(text) => handleCheckPassword(text)}
                    secureTextEntry={seePassword}
                />

                {invalidPassword ? (<Text>{invalidPasswordMsg}</Text>) : (<Text></Text>)}

                <TouchableOpacity
                    style={styles.button}
                    onPress={handleLogin}
                    >
                    <Text>Login</Text>
                </TouchableOpacity>
            </View>
        </TouchableWithoutFeedback>

    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
        width: '100%',
    },
    emailInput: {
        height: 40,
        margin: 12,
        borderWidth: 1,
        padding: 10,
        width: 200,
    },
    passwordInput: {
        height: 40,
        margin: 12,
        borderWidth: 1,
        padding: 10,
        width: 200,
    },
    button: {
        paddingHorizontal: 10,
        paddingVertical: 5,
        borderWidth: 1,
        
    }
});

export default Login;