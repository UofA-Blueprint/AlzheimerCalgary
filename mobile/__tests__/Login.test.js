import React from 'react'
import renderer from 'react-test-renderer'
import Login from '../views/Login'

test('renders correctly', () => {
  const tree = renderer.create(<Login />).toJSON()
  expect(tree).toMatchSnapshot()
})
