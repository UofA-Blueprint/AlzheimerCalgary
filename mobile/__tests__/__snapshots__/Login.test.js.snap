// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text />
  <TextInput
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="Email"
    style={
      {
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
        "width": 200,
      }
    }
    value=""
  />
  <Text />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      {
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
        "width": 200,
      }
    }
    value=""
  />
  <Text />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text>
      Login
    </Text>
  </View>
</View>
`;
